crates/json-abi/src/item.rs

```diff
@@ -356,7 +358,95 @@ impl AbiItem<'_> {
     }
 }
 
+impl FromStr for Constructor {
+    type Err = ParserError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::parse(s)
+    }
+}
+
+impl Constructor {
+    /// Parses a Solidity event signature string:
```

Comment:
```suggestion
    /// Parses a Solidity constructor string:
```

Constructor does not take anonymous